### PR TITLE
Fixes client not sending pings when server continuosly pushes messages

### LINKED ISSUE #336: Client Connection cannot detect idle connection to send ping and server shutdown connection
Client connection should detect that a connection is reading but not sending any data to server to detect heartbeat idle case.
Current implementation checks only last read time which is not enough for reading only connections like just simple listener use case.

Issue from java client:  https://github.com/hazelcast/hazelcast/issues/13576
Fix for java client       :  https://github.com/hazelcast/hazelcast/pull/13577

### PR DESCRIPTION
**Pull request by mustafaiman**
Fixes #336

### CODE REVIEW

File: test/heartbeat/HeartbeatFromServerTest.js [hunk 1]
```diff
@@ -63,7 +65,9 @@ describe('Heartbeat', function () {
         }).then(function () {
             return RC.startMember(cluster.id);
         }).then(function (member2) {
-            simulateHeartbeatLost(client, member2.host + ':' + member2.port, 2000);
+            return memberAddedPromise.promise.then(function () {
```
Comment (mustafaiman) [inline test/heartbeat/HeartbeatFromServerTest.js]: fixes https://github.com/hazelcast/hazelcast-nodejs-client/issues/347

File: test/heartbeat/HeartbeatFromClientTest.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        return RC.createCluster(null, serverConfig).then(function (resp) {
+            cluster = resp;
+        });
+    });
+
+    afterEach(function () {
+        return RC.shutdownCluster(cluster.id);
+    });
+
+    it('client sends heartbeat periodically even when server continuously pushing messages', function () {
```
Comment (SaitTalhaNisanci) [inline test/heartbeat/HeartbeatFromClientTest.js]: `continuously pushes messages`

File: test/heartbeat/HeartbeatFromClientTest.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            client1.getConnectionManager().on('connectionClosed', function () {
+                connectionClosedEventCount++;
+            });
+            return HazelcastClient.newHazelcastClient(clientConfig);
+        }).then(function (c) {
+            client2 = c;
+            return client1.getMap(MAP_NAME);
+        }).then(function (m) {
+            mapFromClient1 = m;
+            return mapFromClient1.addEntryListener({
```
Comment (SaitTalhaNisanci) [inline test/heartbeat/HeartbeatFromClientTest.js]: if there is no op why is this being added?

Comment (mustafaiman) [reply test/heartbeat/HeartbeatFromClientTest.js]: We try to get the server to send messages to this client even though this client does not send any messages to the server. Having registered a listener, server sends updates to client all the time.

**Pull request merged**